src/main/java/org/myrobotlab/service/Serial.java

```diff
@@ -399,14 +399,15 @@ public void open(String inPortName, int rate, int dataBits, int stopBits, int pa
       return;
     }
 
-    if (inPortName.toLowerCase().startsWith("tcp://")) {
+    if (inPortName != null && inPortName.toLowerCase().startsWith("tcp://")) {
       try {
         info("connecting tcp");
         connectTcp(inPortName);
         return;
       } catch (Exception e) {
         // not a big fan of re-throwing exceptions,
         // but I'll make an exception here
+        error("connect ctp threw");

```

Comment:
there's a typo in this error / log message  "ctp" should be "tcp"

Reply:
ahahah ... 
my work ctp is common acronym "common telematics platform" ... muscle memory ..
fixed